Add patch to fix spawn rates and add customization options

RodneyMKay opened the pull request:
Fix spawn rates and add customization options

This patch intends to fix the spawn rates for water and ambient mobs. This has been broken by Bukkit 1.13 because of unclear terminology between animals and peaceful mobs. The terminology used in this patch is that a peaceful mob is everything that is not hostile and an animal is just sheep, pigs, etc. In vanilla all mobs except animals spawn every tick, but Bukkit forced all peaceful mobs to spawn at the animal cycle, which is 400x slower than their normal tick delay per spawn, which is 1.

This patch adds obfuscation helpers to EnumCreatureType to prevent further confusion and adds configuration options to override the Bukkit "ticks-per-spawn" and replace them by customly definable ones.

Note that this bug has already been fixed in Bukkit 1.14.

Commit:
Add patch to fix spawn rates and add customization options

electronicboy commented on Spigot-Server-Patches/0447-Fix-spawn-rates-and-add-customization-options.patch:
```diff
@@ -0,0 +1,146 @@
+ 
+     public boolean matches(Entity entity) { return innerClass().isAssignableFrom(entity.getClass()); } // Paper
+     public Class<? extends IAnimal> innerClass() { return this.a(); } // Paper - OBFHELPER
++
```
OBFHelpers should be placed inline with what they're tryna deobf, these fields have actually changed since this patch was created

RodneyMKay replied:
Sorry, should be fixed now

RodneyMKay committed:
Include new patches

	renamed:    Spigot-Server-Patches/0447-Fix-spawn-rates-and-add-customization-options.patch -> Spigot-Server-Patches/0449-Fix-spawn-rates-and-add-customization-options.patch

Commit:
Moved obfhelpers and adjusted to new patches

Commit:
Move obfhelpers again for extra clarity
